src/Plugins/WebSocket.jslib

```diff
@@ -18,6 +18,46 @@ mergeInto(LibraryManager.library, {
         manager.callbacks.error = errorCallback;
     },
 
+    WebSocket_Generate_Token: function(baseUriPtr, uriPtr, authTokenPtr, callbackPtr) {
```

Comment:
Instead of having this be a separate function I would just do this by default in the connection setup logic. If the user passes an authentication token we should create the short-lived auth token automatically and then pass that as a query parameter instead of exposing a separate function.